src/h5web/App.test.tsx

```diff
@@ -1,23 +1,134 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import {
+  render,
+  RenderResult,
+  fireEvent,
+  waitFor,
+} from '@testing-library/react';
 import App from './App';
 import MockProvider from './providers/mock/MockProvider';
 
 const DOMAIN = 'bsa_002_000-integrate-sub';
 
-test('renders with root group selected', async () => {
-  const { findByText, findByRole } = render(
+function renderApp(): RenderResult {
+  return render(
     <MockProvider domain={DOMAIN}>
       <App />
     </MockProvider>
   );
+}
+
+test('renders with root group selected', async () => {
+  const { findByText, getByRole } = renderApp();
 
   const noVisText = await findByText('Nothing to visualize');
-  expect(noVisText).toBeInTheDocument();
+  expect(noVisText).toBeVisible();
```

Comment:
`toBeVisible` is better than `toBeInTheDocument` as it checks that the element isn't hidden with CSS or by other means. That being said, it's also more computationally heavy as it goes through all the parents to make sure none of them are hidden, so I wouldn't recommend using it every time.